src/main/java/org/tron/core/config/args/Args.java

```diff
@@ -557,8 +511,9 @@ private static void externalIp(final com.typesafe.config.Config config) {
         .getString("node.discovery.external.ip").trim().isEmpty()) {
       if (INSTANCE.nodeExternalIp == null) {
         logger.info("External IP wasn't set, using checkip.amazonaws.com to identify it...");
-        try (BufferedReader in = new BufferedReader(new InputStreamReader(
-            new URL("http://checkip.amazonaws.com").openStream()))) {
+        try {
+          BufferedReader in = new BufferedReader(new InputStreamReader(
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use try-with-resources or close this "BufferedReader" in a "finally" clause. ([squid:S2095](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2095))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=862&issues=AWOmY5GrlfhLAgo0Gw_K&open=AWOmY5GrlfhLAgo0Gw_K)

[ISSUE_KEY]: AWOmY5GrlfhLAgo0Gw_K